go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -348,10 +348,11 @@ func (m *MasterController) setupHybridLRPolicySharedGw(nodeSubnets []*net.IPNet,
 			Match:    matchStr,
 		}
 
+		// TODO(trozet): evaluate if safe to update all fields
 		if err := libovsdbops.CreateOrUpdateLogicalRouterPolicyWithPredicate(m.nbClient, ovntypes.OVNClusterRouter, &logicalRouterPolicy, func(item *nbdb.LogicalRouterPolicy) bool {
 			return item.Priority == logicalRouterPolicy.Priority &&
 				item.ExternalIDs["name"] == logicalRouterPolicy.ExternalIDs["name"]
-		}); err != nil {
+		}, &logicalRouterPolicy.Nexthops, &logicalRouterPolicy.Match); err != nil {
 			return fmt.Errorf("failed to add policy route '%s' for host %q on %s , error: %v", matchStr, nodeName, ovntypes.OVNClusterRouter, err)
```

Comment:
This is equivalent, as we were updating constant fields and neither NextHops or Match could be empty.